crates/config/src/signing.rs

```diff
@@ -0,0 +1,109 @@
+use figment::Figment;
+use figment::providers::{Format, Json, Toml};
+use serde::{Deserialize, Serialize};
+use serde_keys::Bn254SecretKey;
+use shellexpand::tilde;
+use std::num::NonZeroU16;
+use std::path::PathBuf;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct SecretKeyConfig {
+    pub node_id: NonZeroU16,
+    pub secret_key: Bn254SecretKey,
+    pub t: NonZeroU16,
+    pub n: NonZeroU16,
+}
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct GroupConfig {
+    pub nodes: Vec<NodeConfig>,
+}
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct UnvalidatedGroupConfig {
+    pub nodes: Vec<NodeConfig>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct NodeConfig {
+    /// Node identifier used in the threshold scheme
+    pub node_id: NonZeroU16,
+
+    /// BN254 public key of the node
+    #[serde(with = "utils::serialize::point::base64")]
+    pub bls_pk: ark_bn254::G2Affine,
+
+    /// Libp2p peer address
+    pub address: libp2p::Multiaddr,
+
+    /// Peer id
+    pub peer_id: libp2p::PeerId,
+}
+
+impl SecretKeyConfig {
+    pub fn from_path_str(path: impl AsRef<str>) -> anyhow::Result<Self> {
+        Self::from_path(PathBuf::from(tilde(&path).as_ref()))
+    }
+    pub fn from_path(path: PathBuf) -> anyhow::Result<Self> {
+        let c: SecretKeyConfig = Figment::new()
+            .merge(Toml::file(&path))
+            .merge(Json::file(&path))
+            .extract()?;
+        Ok(c)
+    }
+}
+
+impl GroupConfig {
```

Comment:
Nit, but we should decide on terminology at some point, group vs committee, etc.